prow/tide/tide.go

```diff
@@ -217,7 +217,7 @@ func NewController(ghcSync, ghcStatus github.Client, prowJobClient prowv1.ProwJo
 		opener:         opener,
 		path:           statusURI,
 	}
-	go sc.run()
+	sc.run()
```

Comment:
The intent of the status controller _is_ to run indefinitely and not to finish, which is why this is run in a separate (non-blocking) routine. I don't think what you noticed is a bug, unfortunately.

Reply:
Thanks a lot for looking into this. My only concern is that `go sc.run()` will not complete at all and that the function would exit out early.

Consider something like this:

```
func main() {
     go func() {
         time.Sleep(5)
         fmt.Println("First")
    }()
    fmt.Println("Second")
}
```

If I run this, only "Second" would print out and the first goroutine would short circuit. My understanding could be off though.

Reply:
Goroutines will continue to execute after the function that created them returns, your example is a special case where the program itself exits (so no Go anything will run any longer)

Reply:
I decided to test this out and I'm not sure if I'm seeing what you're saying. Take this example:

```
func first() {
    go func() {
        time.Sleep(10)
        fmt.Println("in first goroutine")
    }()
    fmt.Println("outside of first goroutine")
}

func main() {
    first()
    time.Sleep(500)
    fmt.Println("existing main")
}
```

When I run this, I 9/10 times I get:

```
outside of first goroutine
exiting main
```

I figure that if the goroutine did not terminate when the function finished running, I would also see "in first goroutine" printed because of the differences in the time interval.

Can you show me what I'm missing? Thanks again for reviewing this.

Reply:
You still need to block on the routines somehow:

https://play.golang.org/p/Xusa-cwcZJ

Reply:
(or you're not guaranteeing that you wait until they run)